src/metabase-report-notifier/src/main/java/com/champtitles/metabasereportexecutor/notifier/App.java

```diff
@@ -2,15 +2,114 @@
 
 import com.amazonaws.services.lambda.runtime.Context;
 import com.amazonaws.services.lambda.runtime.RequestHandler;
-import com.amazonaws.services.lambda.runtime.events.S3Event;
+import com.amazonaws.services.lambda.runtime.events.SNSEvent;
+import com.champtitles.metabasereportexecutor.executor.KmsDecrypt;
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-public class App implements RequestHandler<S3Event, Void> {
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.NoSuchElementException;
+
+public class App implements RequestHandler<SNSEvent, Void> {
+
+    private static final Logger logger = LoggerFactory.getLogger(App.class.getName());
+    private static final String awsRegion = System.getenv("AWS_REGION");
+    private static final String bucket = System.getenv("BUCKET");
+    private static final String smtpHost = System.getenv("SMTP_HOST");
+    private static final String smtpPort = System.getenv("SMTP_PORT");
+    private static final String smtpUser = System.getenv("SMTP_USER");
+    private static final String smtpPasswordKms = System.getenv("SMTP_PASSWORD_KMS");
+    private static final String fromAddress = System.getenv("FROM_ADDRESS");
+    private static final String recipients = System.getenv().getOrDefault("RECIPIENTS", "test@example.com");
+    private static final String metabaseCardId = System.getenv().getOrDefault("METABASE_CARD_ID", "1");
+    private static final String name = System.getenv().getOrDefault("NAME", "Test");
+    private static final String sizeLimitBytes = System.getenv().getOrDefault("SIZE_LIMIT_BYTES", "26214400");
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private static final JsonPointer objectKeyPtr = JsonPointer.compile("/Records/0/s3/object/key");
+    private final S3Reader s3Reader;
+    private final EmailSender emailSender;
+
+    public App() {
+        this(new S3Reader(bucket), new EmailSender(smtpHost, smtpPort, smtpUser, new KmsDecrypt((awsRegion)).decrypt(smtpPasswordKms), fromAddress));
+    }
+
+    App(S3Reader s3Reader, EmailSender emailSender) {
+        this.s3Reader = s3Reader;
+        this.emailSender = emailSender;
+    }
 
     @Override
-    public Void handleRequest(final S3Event event, final Context context) {
-        System.out.println("hello world");
-        System.out.println(event.toString());
+    public Void handleRequest(SNSEvent snsEvent, Context context) {
+
+        for (SNSEvent.SNSRecord snsRecord : snsEvent.getRecords()) {
+            String s3Key = parseS3Key(snsRecord.getSNS().getMessage());
+            byte[] data = s3Reader.downloadXlsx(s3Key);
+            logger.info("downloaded {} bytes", data.length);
```

Comment:
If the transaction context is not assuredly evident from other logs and how lambda logs, then need some useful context here -- card #?